### ISSUE TITLE
Multiline Comment: the 2nd line can be wrongly indented 

### ISSUE THREAD

**Issue by derekwallace** [microsoft/pylance-release#5488]
Type: <b>Bug</b>

# Behaviour
## Expected vs. Actual

It is expected that if i start a comment and hit return that the next line starts at the samw colum that the previous started.
What actually happens is the next line starts at the indent level of the last line of code before the commetn

It used to work well. As far as i recall
1. the next line would auto start with comment char #
2. it would maintain alignement of the # char with the prevous line

WIndows
VsCode 1.851
Python and Pylance plugins

This is what happens. when you hit return after the first comment line it jumps to a column that lines up with the previous print() statement.

![comment](https://github.com/microsoft/vscode-python/assets/5574230/735f3ab6-f89c-4260-9f0a-0c69ad63a4ab)

This is what it should look like
![image](https://github.com/microsoft/vscode-python/assets/5574230/21bdd7eb-68d2-44e0-98cd-f5926adbb552)

Extension version: 2024.0.1
VS Code version: Code 1.85.1 (0ee08df0cf4527e40edc9aa28f4b5bd38bbff2b2, 2023-12-13T09:49:37.021Z)
OS version: Windows_NT x64 10.0.19044
Modes:
Remote OS version: Linux x64 3.10.0-1160.31.1.el7.x86_64
Remote OS version: Linux x64 3.10.0-1160.95.1.el7.x86_64

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|11th Gen Intel(R) Core(TM) i7-1185G7 @ 3.00GHz (8 x 1805)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|undefined|
|Memory (System)|15.40GB (1.38GB free)|
|Process Argv|--crash-reporter-id b37c34d7-f099-4c80-8321-342fc4151635|
|Screen Reader|no|
|VM|0%|

|Item|Value|
|---|---|
|Remote|SSH: vividus01|
|OS|Linux x64 3.10.0-1160.31.1.el7.x86_64|
|CPUs|Intel Core Processor (Skylake) (4 x 2900)|
|Memory (System)|83.22GB (68.64GB free)|
|VM|0%|

|Item|Value|
|---|---|
|Remote|SSH: vivideudc_VM|
|OS|Linux x64 3.10.0-1160.95.1.el7.x86_64|
|CPUs|AMD EPYC 7713P 64-Core Processor (8 x 2000)|
|Memory (System)|70.63GB (44.63GB free)|
|VM|100%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vspor879:30202332
vspor708:30202333
vspor363:30204092
vscoreces:30445986
vscod805:30301674
binariesv615:30325510
vsaa593cf:30376535
py29gd2263:30899288
c4g48928:30535728
azure-dev_surveyone:30548225
vscrp:30673768
962ge761:30959799
pythongtdpath:30769146
welcomedialogc:30910334
pythonidxpt:30866567
pythonnoceb:30805159
asynctok:30898717
pythontestfixt:30902429
pythonregdiag2:30936856
pyreplss1:30897532
pythonmypyd1:30879173
pythoncet0:30885854
pythontbext0:30879054
accentitlementsc:30887149
dsvsc016:30899300
dsvsc017:30899301
dsvsc018:30899302
pydisww1:30959800
7j2b6412:30964151
e3gdj431:30958358

```

</details>

**Comment by christo-olivier-sp** [comment]
Just wanted to add that I am experiencing the same issue and that disabling Pylance stopped this behaviour (obviously not a solution). 

I do not consider this behaviour a feature, it is a bug. If I hit return/enter to write the next line of a comment I expect the cursor to end up at the column that the previous comment line was.

I dug into the settings for "editor.autoindent" and saw that it was set to "full" but mentioned _(Modified elsewhere)_. Two languages seem to have their own settings for this YAML and Compose. I set the setting to apply to all profiles and then this behaviour stopped.

**Comment by heejaechang** [comment]
another approach we could take is leaving things same as original, but when user type '#', align it to the previous comment's indentation?

**Comment by StellaHuang95** [comment]
This issue has been fixed in prerelease version 2024.3.102, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#20243102-27-March-2024-PreRelease)
